pkg/apiclient/http1/facade.go

```diff
@@ -65,12 +67,15 @@ func (h Facade) EventStreamReader(in interface{}, path string) (*bufio.Reader, e
 	req.Header.Set("Accept", "text/event-stream")
 	req.Header.Set("Authorization", h.authorization)
 	log.Debugf("curl -H 'Accept: text/event-stream' -H 'Authorization: ******' '%v'", u)
-	client := &http.Client{
-		Transport: &http.Transport{
-			TLSClientConfig: &tls.Config{
-				InsecureSkipVerify: h.insecureSkipVerify,
+	client := h.httpClient
+	if h.httpClient == nil {
+		client = &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+					InsecureSkipVerify: h.insecureSkipVerify,
+				},
 			},
-		},
+		}
```

Comment:
Not sure why we need to do this, I am not entirely familiar with this part of the codebase, what are your opinions on ensuring that client is never null? 

This can be achieved by moving this check to NewFacade.

I am not sure why this wasn't done before.